### ISSUE TITLE
/PF gives "Invalid parameter" error

### ISSUE THREAD

**Issue by JoanEliot** [ildar-shaimordanov/psubst#21]
Whereas /P works fine. This is in an elevated command prompt, in case that matters

C:\util>psubst /PF K: C:\z\K
psubst: Invalid parameter: /PF

**Comment by ildar-shaimordanov** [comment]
That's correct because the order of the arguments and options is mandatory. Run the tool with `/?` option and to learn more about usage. The correct usage is as follows:

```
PSUBST [drive1: [drive2:]path] [/P | /PF]
```

**Comment by JoanEliot** [comment]
Sorry! I missed your reply before I answered. I figured out I had the argument in the wrong place, and then ran into a new issue as above. But as for the order or arguments point, for me I get no error message with pubst /P so maybe an update to recognize that also as an out-of-order parameter?

**Comment by ildar-shaimordanov** [comment]
The command `psubst /P K: C:\z\K` doesn't throw any issues because it's executed as this one `psubst /P` (ignoring everything after `/p`).

Regarding the second issue `Unable to add persistently SUBSTed drive`. I reviewed the code and didn't find any reasons to throw it with the `/pf` option. This could happen with the `/p` option only (and without elevated privileges).

Can you remove the entry with the command `psubst K: /d /pf`, then repeat the creation command `psubst K: C:\z\K /PF`? Let me know more details.

**Comment by JoanEliot** [comment]
Results:
`>psubst K: /d /pf
psubst: Drive not SUBSTed persistently
>k:
K:\>`

'>psubst K: C:\z\K /PF
psubst: Unable to add persistently SUBSTed drive'

**Comment by ildar-shaimordanov** [comment]
I need to understand or become closer in understanding what's wrong in your situation. Would you be perform the following steps:
1. open RegEdit application and proceed the path `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\DOS Devices`; find the `K:` value and delete it
2. Run `psubst /p`
3. Run `psubst K: C:\z\K /PF`
4. Run `psubst /p` again

Share results here.

**Comment by JoanEliot** [comment]
Sorry for not replying earlier. Still pretty busy so can't promise prompt feedback but here you go—

Results (tempfile listing -- note that there is no entry for K )

```
"AUX"="\\DosDevices\\COM1"
"CON"="\\Device\\ConDrv\\Console"
"CONIN$"="\\Device\\ConDrv\\CurrentIn"
"CONOUT$"="\\Device\\ConDrv\\CurrentOut"
"MAILSLOT"="\\Device\\MailSlot"
"NUL"="\\Device\\Null"
"PIPE"="\\Device\\NamedPipe"
"PRN"="\\DosDevices\\LPT1"
"Silos"="\\Silos"
"UNC"="\\Device\\Mup"

>psubst K: C:\z\K /PF
psubst: Unable to add persistently SUBSTed drive
```

**Comment by JoanEliot** [comment]
Sorry for not replying earlier. Still pretty busy so can't promise prompt feedback but here you go—

Results (tempfile listing -- note that there is no entry for K )

```
"AUX"="\\DosDevices\\COM1"
"CON"="\\Device\\ConDrv\\Console"
"CONIN$"="\\Device\\ConDrv\\CurrentIn"
"CONOUT$"="\\Device\\ConDrv\\CurrentOut"
"MAILSLOT"="\\Device\\MailSlot"
"NUL"="\\Device\\Null"
"PIPE"="\\Device\\NamedPipe"
"PRN"="\\DosDevices\\LPT1"
"Silos"="\\Silos"
"UNC"="\\Device\\Mup"

>psubst K: C:\z\K /PF
psubst: Unable to add persistently SUBSTed drive
```

**Comment by ildar-shaimordanov** [comment]
@JoanEliot, I guess, the user you work with doesn't have writing permissions in the particular subkey. Let's try one more attempt.

1. Download the file `sudo.bat` from this link: https://raw.githubusercontent.com/ildar-shaimordanov/cmd.scripts/master/unix-like/sudo.bat. Save it next to `psubst.bat`
2. Run it. It will open a new console window (titled as "Administrator")
3. In this new window run the command `psubst K: C:\z\K /p`
4. Check the list of substed drives with `psubst /p`

Capture the results of all commands you execute in this attempt and share.
